### ISSUE TITLE
L254.transportation_UCD only produces a single scenario at a time

### ISSUE THREAD

**Issue by pralitp** [JGCRI/gcamdata#696]
The UCD transportation chunk is currently only able to produce a single scenario at a time unlike all other chunks which generate inputs for **all SSPs**.  Instead it checks a configuration variable to determine which scenario to generate.  If a user wants to generate another they need to change the value and re-run.  For the time being we are continuing with the previous approach but this is something we would like to improve upon.

Also see: #601

**Comment by pkyle** [comment]
Yeah, it's because the transportation code files used to take really long to run. From what I remember, the time-intensive steps were (L154) interpolating all variables in a really big dataset, and (L254) figuring out all of the correct calibration values to read for the many pass-through technologies, which is not an easy thing to do. If that second step is fast now, then we could just run both code chunks in a for loop, looping through the 5 scenarios and writing everything out. If we need to vectorise to save on run time, then things will get a bit more tricky.
Another difference here from the other sectors is that the SSPs were implemented by Gouri, and all of the data that Gouri developed is in the "UCD_transportation_database_SSPx.csv" files, which include both historical data and future assumptions. While only the future assumptions differ by SSP, the historical data is nevertheless available in each, and the existing code uses that in its processing. Only relevant where they (unintentionally) differ.
On that note, there might be a mistake that we never fixed whereby the Middle East's historical aviation energy is mis-allocated between international and domestic, in some but not all of the transportation-SSP databases. I made a Jira and/or Confluence page about it a year or two ago, and might have submitted it as a core model proposal, but the consequences of the error were really too small to worry about, and I'm pretty sure the fix never made it into the core model. That would cause some unexpected differences by SSP in the historical years.
